src/main/java/net/dries007/tfc/objects/items/ceramics/ItemJug.java

```diff
@@ -1,101 +1,129 @@
+import net.minecraft.util.EnumActionResult;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.SoundCategory;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.fluids.BlockFluidBase;
+import net.minecraftforge.fluids.Fluid;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.capability.CapabilityFluidHandler;
+import net.minecraftforge.fluids.capability.IFluidHandler;
+
+public class ItemJug extends ItemFluidStoragePottery
+{
+
+	private static final int MAX_FLUID_AMOUNT = 1000;
+	private static final float BREAK_CHANCE = 0.02f;
+	private static final int USE_DURATION = 32;
+
+	public ItemJug()
+	{
+		super(USE_DURATION);
+	}
+	
+	@Override
+	public void addInformation(ItemStack stack, World worldIn, List<String> tooltip, ITooltipFlag flagIn)
+	{
+		super.addInformation(stack, worldIn, tooltip, flagIn);
+		if(isFilled(stack))
+		{
+			tooltip.add(String.format("Amount: %dmb", MAX_FLUID_AMOUNT));
+		}
+	}
+
+	@Override
+	public EnumActionResult onItemUse(EntityPlayer player, World worldIn, BlockPos pos, EnumHand hand, EnumFacing facing, float hitX, float hitY, float hitZ)
+	{
+		if(!worldIn.isRemote)
+		{
+			ItemStack stack = player.getHeldItem(hand);
+			ItemJug jug = (ItemJug) stack.getItem();
+			IBlockState fluidBlock = worldIn.getBlockState(pos.up());
+			Fluid fluid = null;
+			
+			final Block block = worldIn.getBlockState(pos).getBlock();
+
+			if(!jug.isFilled(stack))
+			{
+				if(block != Blocks.AIR && block.hasTileEntity(worldIn.getBlockState(pos)))
+				{
+					TileEntity te = worldIn.getTileEntity(pos);
+					if(te != null && te.hasCapability(CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, facing))
+					{
+						IFluidHandler cap = (IFluidHandler) te.getCapability(CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, facing);
+						FluidStack fluidStack = cap.drain(MAX_FLUID_AMOUNT, false);
+						fluid = fluidStack == null ? null : fluidStack.getFluid();
+						if(fluid != null)
+						{
+							// Drain from the fluid storage into the jug
+							cap.drain(MAX_FLUID_AMOUNT, true);
+						}
+					}
+				}
+				else if(fluidBlock != null && fluidBlock.getBlock() instanceof BlockFluidBase)
+				{
+					fluid = ((BlockFluidBase) fluidBlock.getBlock()).getFluid();
+				}
+				
+				if(fluid != null)
+				{	
+					jug.setFluid(stack, fluid);
+					jug.setJustFilled(stack, true);
+					return EnumActionResult.FAIL;
+				}
+			}
+			else if(block.hasTileEntity(worldIn.getBlockState(pos)))
+			{
+				TileEntity te = worldIn.getTileEntity(pos);
+				if(te != null && te.hasCapability(CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, facing))
+				{
+					IFluidHandler cap = (IFluidHandler) te.getCapability(CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, facing);
+					FluidStack fluidStack = new FluidStack(jug.getFluid(stack), MAX_FLUID_AMOUNT);
+					if(cap.fill(fluidStack, false) == MAX_FLUID_AMOUNT)

```

Comment:
Jugs are not buckets.
They can be filled only. Emptying the fluid within should destroy it or place a finite block.

Reply:
They can only be filled once and I haven't add the ability to empty the bucket into the world as a source block or something. Should I really do that though? I mean I can destroy it on place of course. Also, there's a small chance of the jug destroyed when you drink though like 2% chance every time you drink.

Reply:
Unless jugs are going to be the "Wooden bucket" of the rewrite they shouldn't transfer fluids either. I guess that's up to Alcatraz to say then.

Reply:
Yeah, probably. You can pretty much transfer liquid into barrels and etc. easily without even a bucket. I guess I need to remove it then if "Wooden bucket" have that feature or else it would be useless

Reply:
Let Alcatraz look it over first. Strictly speaking 1.7.10 if the jug did the same thing as the wooden bucket and the bucket didn't exist, nothing would have changed really. If they held the chance to break on empty perhaps a better option could be made further up the tech line.

Reply:
Okay then. I'll leave it up to the master to decide

Reply:
The jug can now place temporary flowing fluid for a few seconds